CMakeLists.txt

```diff
@@ -1,421 +1,201 @@
+  CARLA_VERSION 
+  ${CARLA_VERSION_MAJOR}.${CARLA_VERSION_MINOR}.${CARLA_VERSION_PATCH}
+)
 
 project (
-    CARLA
-    VERSION ${CARLA_VERSION}
-    LANGUAGES CXX
-    DESCRIPTION "Open-source simulator for autonomous driving research."
-    HOMEPAGE_URL "https://carla.org/"
+  CARLA
+  VERSION ${CARLA_VERSION}
+  LANGUAGES CXX
+  DESCRIPTION "Open-source simulator for autonomous driving research."
+  HOMEPAGE_URL https://carla.org
 )
 
+set (CMAKE_CXX_STANDARD 20)
+set (CMAKE_CXX_STANDARD_REQUIRED ON)
+
 include (CheckCCompilerFlag)
 include (CheckCXXCompilerFlag)
 
-set (CMAKE_CXX_STANDARD 20)
-set (CMAKE_CXX_STANDARD_REQUIRED ON)
-set (CMAKE_EXPORT_COMPILE_COMMANDS ON)
+set (CARLA_WORKSPACE_PATH ${CMAKE_CURRENT_SOURCE_DIR})
+
+list (
+  APPEND
+  CMAKE_MODULE_PATH
+  ${CARLA_WORKSPACE_PATH}/CMake
+)
+
+list (
+  APPEND
+  CMAKE_PREFIX_PATH
+  ${CARLA_WORKSPACE_PATH}/CMake
+)
+
+macro (carla_string_option NAME DESCRIPTION VALUE)
+  set (${NAME} ${VALUE} CACHE STRING ${DESCRIPTION})
+endmacro ()
+
+carla_string_option (
+  CARLA_DEPENDENCIES_PATH
+  "Where to build CARLA dependencies."
+  ${CMAKE_CURRENT_BINARY_DIR}/Dependencies
+)
+
+carla_string_option (
+  CARLA_DEPENDENCIES_INSTALL_PREFIX
+  "Where to install CARLA dependencies."
+  ${CMAKE_CURRENT_SOURCE_DIR}/Install
+)
+
+set (
+  CMAKE_INSTALL_PREFIX
+  ${CARLA_DEPENDENCIES_INSTALL_PREFIX}
+)
 
 option (
-    BUILD_LIBCARLA_CLIENT
-    "Build the LibCarla client."
-    ON)
+  BUILD_CARLA_CLIENT
+  "Build the LibCarla client."
+  ON
+)
 
 option (
-    BUILD_LIBCARLA_SERVER
-    "Build the LibCarla server."
-    ON)
+  BUILD_CARLA_SERVER
+  "Build the LibCarla server."
+  ON
+)
 
 option (
-    ENABLE_LIBCARLA_PYTORCH
-    "Enable LibCarla-PyTorch."
-    OFF)
+  ENABLE_LIBCARLA_PYTORCH
+  "Enable LibCarla-PyTorch."
+  OFF
+)
 
 option (
-    BUILD_PYTHON_API
-    "Build the CARLA Python API."
-    ON)
+  BUILD_PYTHON_API
+  "Build the CARLA Python API."
+  ON
+)
 
 option (
-    ENABLE_LIBCARLA_CLIENT_RSS
-    "Enable RSS components (ad-rss-lib)"
-    OFF)
+  ENABLE_OSM2ODR
+  "Enable OSM2ODR."
+  ${BUILD_PYTHON_API}
+)
 
 option (
-    INSTALL_LIBCARLA
-    "Install LibCarla and all of its dependencies."
-    ON)
+  ENABLE_RSS
+  "Enable RSS components (ad-rss-lib)"
+  OFF
+)
 
 option (
-    BUILD_OSM_WORLD_RENDERER
-    "Build OSM World Renderer."
-    ON)
+  INSTALL_LIBCARLA
+  "Install LibCarla and all of its dependencies."
+  ON
+)
 
 option (
-    ENABLE_RTTI
-    "Enable C++ RTTI."
-    OFF)
+  BUILD_OSM_WORLD_RENDERER
+  "Build OSM World Renderer."
+  ON
+)
 
 option (
-    ENABLE_EXCEPTIONS
-    "Enable C++ Exceptions."
-    OFF)
+  ENABLE_RTTI
+  "Enable C++ RTTI."
+  OFF
+)
 
-set (CARLA_WORKSPACE_PATH ${CMAKE_CURRENT_SOURCE_DIR})
-set (CARLA_BUILD_PATH ${CMAKE_CURRENT_BINARY_DIR})
-if (NOT CARLA_DEPENDENCIES_PATH)
-    set (CARLA_DEPENDENCIES_PATH ${CMAKE_CURRENT_BINARY_DIR}/Dependencies)
+option (
+  ENABLE_EXCEPTIONS
+  "Enable C++ Exceptions."
+  OFF
+)
+
+if (WIN32)
```

Comment:
This if should be not needed anymore as CMake have ways to automaticaly disambiguate library names like `find_package()`and `add_external_library()`